mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -150,11 +150,17 @@ impl S3CrtClient {
     /// object data.
     fn new_request_template(&self, method: &str, bucket: &str) -> Result<S3Message, ConstructionError> {
         let (uri, path_prefix) = self.endpoint.for_bucket(bucket)?;
-        let hostname = uri.host_name();
+        let hostname = uri.host_name().to_str().unwrap();
+        let port = uri.host_port();
+        let hostname_header = if port > 0 {
+            format!("{}:{}", hostname, port)
+        } else {
+            format!("{}", hostname)
```

Comment:
(Sorry, I messed this up, this is making Clippy mad)
```suggestion
            hostname.to_string()
```